From: SeongJae Park <sj@kernel.org>
Date: Thu, 1 Jan 2026 14:56:48 -0800
Subject: [PATCH] mm/damon: document damon_call_control->dealloc_on_cancel
 repeat behavior

damon_call_control->dealloc_on_cancel works only when ->repeat is NULL,
but the behavior is not clearly documented.  DAMON API callers can
understand the behavior only after reading kdamond_call() code.
Document the behavior on the kernel-doc comment of damon_call_control.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 include/linux/damon.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/damon.h b/include/linux/damon.h
index 905100fa51f0..181588972fcb 100644
--- a/include/linux/damon.h
+++ b/include/linux/damon.h
@@ -719,7 +719,7 @@ struct damon_operations {
  * @data:		Data that will be passed to @fn.
  * @repeat:		Repeat invocations.
  * @return_code:	Return code from @fn invocation.
- * @dealloc_on_cancel:	De-allocate when canceled.
+ * @dealloc_on_cancel:	De-allocate when canceled, if @repeat is NULL.
  *
  * Control damon_call(), which requests specific kdamond to invoke a given
  * function.  Refer to damon_call() for more details.
-- 
2.47.3

